agent/php_api.c

```diff
@@ -168,13 +172,23 @@ PHP_FUNCTION(newrelic_notice_error) {
     }
   }
 
-  {
-    char* buf = nr_strndup(errormsgstr, errormsglen);
+  if (!determine) {
```

Comment:
This code seems to be able to be reduced as there are common elements to each possible path - also use an `else` instead of having two separate `if` statements testing logically opposite conditions.

Reply:
Fixed: [66a069d](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/66a069dd06cd14f1dc72f87f940a794d8180053a)